src/Elasticsearch.Net.Aws/Elasticsearch.Net.Aws/SigningHttpMessageHandler.cs

```diff
@@ -0,0 +1,40 @@
+﻿#if NETSTANDARD
+using Amazon;
+using Amazon.Runtime;
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Elasticsearch.Net.Aws
+{
+    class SigningHttpMessageHandler : HttpClientHandler
+    {
+        readonly AWSCredentials _credentials;
+        readonly RegionEndpoint _region;
+        readonly HttpMessageHandler _innerHandler;
+
+        public SigningHttpMessageHandler(AWSCredentials credentials, RegionEndpoint region, HttpMessageHandler innerHandler)
+        {
+            _credentials = credentials;
+            _region = region;
+            _innerHandler = innerHandler;
+        }
+
+        protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            var credentials = await _credentials.GetCredentialsAsync().ConfigureAwait(false);
+            await SignV4Util.SignRequestAsync(new HttpRequestMessageAdapter(request), credentials, _region.SystemName, "es").ConfigureAwait(false);
+            return await _innerHandler.SendAsync(request, cancellationToken).ConfigureAwait(false);
+        }
+
+        protected override void Dispose(bool disposing)
```

Comment:
This one causes Stack overflow exception. 
Dispose method has to be updated as in follow documentation
https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose#implement-the-dispose-pattern-for-a-derived-class

```
protected override void Dispose(bool disposing)
        {
            if(!_disposedValue)
            {
                if(disposing)
                {
                    _innerHandler.Dispose();
                }

                _disposedValue = true;
            }

            base.Dispose(disposing);
        }
```

Reply:
Okay, I pushed 9186e736460948274067ef0c5f852d7980100901 to address it.